### PR TITLE
feat: add support for web transports

### LINKED ISSUE #190: Feature: WebTransports
Add support for WebTransports https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-02.html

### LINKED ISSUE #193: Panic when connecting to old servers (or WebTransports)
When hitting a server which reports that it wants to use WebTransports signalR brings down the whole process due to a programming error. This error can't be gracefully handled as it is a panic within a Goroutine singalR spawns.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1f912e1]

goroutine 748 [running]:
github.com/philippseith/signalr.(*client).sendHandshakeRequest(0xc00062e640)
	github.com/philippseith/signalr@v0.6.3/client.go:575 +0x41
github.com/philippseith/signalr.(*client).processHandshake(0x2711400?)
	github.com/philippseith/signalr@v0.6.3/client.go:568 +0x18
github.com/philippseith/signalr.(*client).setupConnectionAndProtocol.func1(0xc00062e640)
	github.com/philippseith/signalr@v0.6.3/client.go:299 +0x137
github.com/philippseith/signalr.(*client).setupConnectionAndProtocol(0x7f6384d16d00?)
	github.com/philippseith/signalr@v0.6.3/client.go:305 +0x13
github.com/philippseith/signalr.(*client).run(0xc00062e640)
	github.com/philippseith/signalr@v0.6.3/client.go:230 +0x1c
github.com/philippseith/signalr.(*client).Start.func1()
	github.com/philippseith/signalr@v0.6.3/client.go:188 +0x11d
created by github.com/philippseith/signalr.(*client).Start in goroutine 43
	github.com/philippseith/signalr@v0.6.3/client.go:171 +0xa6
```

Edit: the stack above came actually when trying to connect to a old server. However in both cases the panic is the same

### PR DESCRIPTION
**Pull request by sruehl**
resolves https://github.com/philippseith/signalr/issues/190
resolves https://github.com/philippseith/signalr/issues/193

### CODE REVIEW

**Comment by philippseith** [comment]
Nice!
Were you able to verify it with any third party client/server? AFAIK aspnetcore did not implement it yet: https://github.com/dotnet/aspnetcore/issues/39583

**Comment by sruehl** [comment]
At the moment I'm in the process of verification, hence the PR is still in draft. I'll update here once I know that it works

File: httpconnection.go [hunk 1]
```diff
@@ -126,7 +127,14 @@ func NewHTTPConnection(ctx context.Context, address string, options ...func(*htt
 	var conn Connection
 	switch {
 	case negotiateResponse.hasTransport("WebTransports"):
-		// TODO
```
Comment (sruehl) [inline httpconnection.go:129]: This way the way I noticed btw as this lead to a nasty nil pointer at a completely unrelated place (connection nil and error nil at the same time).

**Comment by sruehl** [comment]
Also I wonder if we should merge it in this state and fix potential issues later or temporary return a proper error on master?

**Comment by sruehl** [comment]
And regarding the failing Unit test. I have no idea what is going on there...

**Comment by sruehl** [comment]
> Also I wonder if we should merge it in this state and fix potential issues later or temporary return a proper error on master?

@philippseith I prepared a PR for option 2 https://github.com/philippseith/signalr/pull/192.

**Comment by sruehl** [comment]
seems that Unit test failure is unrelated to this PR and is caused by https://github.com/philippseith/signalr/commit/35b4977f8fe6e61cdb111ae3977ad717e2bfe4ec

**Comment by sruehl** [comment]
I rebased this on #192

**Comment by sruehl** [comment]
This PR should no be safe to merge as it fixes two issues:

1. When Server responds WebTransports the client should stop panic (#192 at least stops the panic but you still get a error when Server responds with this Transport type)
2. WebTransports is only used in this state when the Client explicitly configures the Transport via ClientOptions

**Pull request merged**